osidb/models.py

```diff
@@ -1788,6 +1789,55 @@ class Meta:
         )
 
 
+class FlawReferenceManager(ACLMixinManager, TrackingMixinManager):
+    """flawreference manager"""
+
+    @staticmethod
+    def create_flawreference(flaw, _type, url, **extra_fields):
+        """return a new flawreference or update an existing flawreference without saving"""
+        try:
+            flawreference = FlawReference.objects.get(flaw=flaw, type=_type, url=url)
+            for attr, value in extra_fields.items():
+                setattr(flawreference, attr, value)
+            return flawreference
+
+        except ObjectDoesNotExist:
+            return FlawReference(flaw=flaw, type=_type, url=url, **extra_fields)
+
+
+class FlawReference(TrackingMixin, ACLMixin):
+    """Model representing flaw references"""
+
+    class FlawReferenceType(models.TextChoices):
+        """allowable references"""
+
+        VULN_RESPONSE = "VULN_RESPONSE"
+        EXTERNAL = "EXTERNAL"
```

Comment:
I assume the name `vuln response` is historical: to me it means nothing, what differentiates these two types? should we rename them?

Reply:
We should at least document the implications of each type in the code itself

Reply:
Sure, but what does `vuln_response` mean? What's the difference between an `external` and `vuln_response` link?

Reply:
I checked that links on sfm2/cve page, and it seems we have to differentiate between them. If I understand it correctly, `vuln_response` represents "Security Bulletin" and `external` represents "External References" in e.g. the following [cve page](https://access.redhat.com/security/cve/cve-2022-0847). 

So I added a description of these links, and I am also going to add validations for `vuln_response`. It should start with https://accesss.redhat.com/, and IMHO every flaw/cve should have only one `vuln_response` link (already discussed with @osoukup). Do you agree with it, @Elkasitu?

Reply:
Maybe checking that the internal / vuln_response URL is a `redhat.com` subdomain is good enough and a bit more future-proof?

Reply:
@Elkasitu I would change `vuln_response` to `security_bulletin_url` and `external_references` leave as it is. Do you agree?

Reply:
ok let's keep it as `security_bulletin` and `external` then

Reply:
Updated.